### PR TITLE
Check already installed version

### PR DESCRIPTION
**Pull request by hendrikebbers**
Dialog API refactored
Dialog can now be used EDT independend
New Dialog Check for installing same version twice

### CODE REVIEW

File: openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java [hunk 1]
```diff
@@ -76,20 +73,53 @@ public LocalJavaRuntime getRuntime(final VersionString versionString, final URL
             final RemoteJavaRuntime remoteJavaRuntime = RemoteRuntimeManager.getInstance().getBestRuntime(versionString, serverEndpoint, vendor, os)
                     .orElseThrow(() -> new RuntimeException("Can not provide or find runtime for version '" +
                             versionString + "', vendor '" + vendor + "' and operating system '" + os + "'"));
+
+            //CHECK IF WE ALREADY HAVE THE SAME RUNTIME (DEACTIVATED & MANAGED) LOCALLY
```
Comment (sclassen) [inline openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java]: If you move this new code block into `installRemoteRuntime()` then you do not need to duplicate it.

File: openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java [hunk 2]
```diff
@@ -76,20 +73,53 @@ public LocalJavaRuntime getRuntime(final VersionString versionString, final URL
...
         }
     }
 
+    private LocalJavaRuntime findMatchingDeactivatedAndManagedLocalRuntime(final JavaRuntime runtime) {
+        return LocalRuntimeManager.getInstance().getAll().stream()
+                .filter(l -> !l.isActive())
+                .filter(l -> l.isManaged())
+                .filter(l -> Objects.equals(l.getVersion(), runtime.getVersion()))
+                .filter(l -> Objects.equals(l.getVendor(), runtime.getVendor()))
+                .findAny()
```
Comment (sclassen) [inline openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java]: use `findFirst()` to be reproducable

File: openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/DialogWithResult.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import java.awt.BorderLayout;
+import java.util.Arrays;
+
+public class DialogWithResult<R> extends JDialog {
+
+    private R result;
+
+    public DialogWithResult(final String title, final String message,DialogButton<R>... buttons) {
```
Comment (sclassen) [inline openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/DialogWithResult.java]: space and final before `DialogButton<R>`

File: openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/DialogWithResult.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        this(title, message, new ImageIcon(Images.QUESTION_64_URL), buttons);
+    }
+
+    public DialogWithResult(final String title, final String message, final ImageIcon imageIcon, DialogButton<R>... buttons) {
```
Comment (sclassen) [inline openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/DialogWithResult.java]: final

**Comment by hendrikebbers** [comment]
@sclassen since I now created a second Yes/No dialog the refactoring just made sense ;)

File: openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/DialogWithResult.java (hunk 4 above)
Comment (hendrikebbers) [reply openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/DialogWithResult.java]: Done

File: openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/DialogWithResult.java (hunk 3 above)
Comment (hendrikebbers) [reply openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/DialogWithResult.java]: Done

File: openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java (hunk 2 above)
Comment (hendrikebbers) [reply openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java]: Done

File: openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java (hunk 1 above)
Comment (hendrikebbers) [reply openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java]: good point :)

**Pull request merged**